atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbReadTable.java

```diff
@@ -16,45 +16,174 @@
+        if (columns.noColumnsSelected()) {
+            return ClosableIterators.emptyImmutableClosableIterator();
+        } else if (isSingleton(rows)) {
+            byte[] row = Iterables.getOnlyElement(rows);
+            return run(queryFactory.getLatestRowQuery(row, ts, columns, includeValues));
+        } else {
+            return run(queryFactory.getLatestRowsQuery(rows, ts, columns, includeValues));
+        }
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getAllRows(
+            Iterable<byte[]> rows, ColumnSelection columns, long ts, boolean includeValues) {
+        if (columns.noColumnsSelected()) {
+            return ClosableIterators.emptyImmutableClosableIterator();
+        } else if (isSingleton(rows)) {
+            byte[] row = Iterables.getOnlyElement(rows);
+            return run(queryFactory.getAllRowQuery(row, ts, columns, includeValues));
+        } else {
+            return run(queryFactory.getAllRowsQuery(rows, ts, columns, includeValues));
+        }
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getLatestCells(Map<Cell, Long> cells, boolean includeValue) {
+        if (cells.size() == 1) {
+            Map.Entry<Cell, Long> onlyEntry = Iterables.getOnlyElement(cells.entrySet());
+            return run(queryFactory.getLatestCellQuery(onlyEntry.getKey(), onlyEntry.getValue(), includeValue));
+        } else {
+            return run(queryFactory.getLatestCellsQuery(cells.entrySet(), includeValue));
+        }
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getAllCells(Iterable<Cell> cells, long ts, boolean includeValue) {
+        if (isSingleton(cells)) {
+            Cell cell = Iterables.getOnlyElement(cells);
+            return run(queryFactory.getAllCellQuery(cell, ts, includeValue));
+        } else {
+            return run(queryFactory.getAllCellsQuery(cells, ts, includeValue));
+        }
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getRange(RangeRequest range, long ts, int maxRows) {
+        FullQuery query = queryFactory.getRangeQuery(range, ts, maxRows);
+        AgnosticLightResultSet results = conns.get().selectLightResultSetUnregisteredQuery(
+                query.getQuery(), query.getArgs());
+        results.setFetchSize(maxRows);
+        return ClosableIterators.wrap(results.iterator(), results);
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getRowsColumnRangeCounts(
             List<byte[]> rows,
             long ts,
-            ColumnRangeSelection columnRangeSelection);
+            ColumnRangeSelection columnRangeSelection) {
+        FullQuery query = queryFactory.getRowsColumnRangeCountsQuery(rows, ts, columnRangeSelection);
+        AgnosticLightResultSet results = conns.get()
+                .selectLightResultSetUnregisteredQuery(query.getQuery(), query.getArgs());
+        results.setFetchSize(Math.max(rows.size(), MAX_ROW_COLUMN_RANGES_FETCH_SIZE));
```

Comment:
yeah, looks like its been set this way since #724. I'm fine with that being separate PR.